src/libs/actions/User.js

```diff
@@ -65,6 +65,56 @@ function getBetas() {
     });
 }
 
+/**
+ * Fetch the public domain info for the current user (includes secondary logins).
+ *
+ * This API is a bit weird in that it sometimes depends on information being cached in bedrock.
+ * If the info for the domain is not in bedrock, then it creates an asynchronous bedrock job to gather domain info.
+ * If that happens, this function will automatically retry itself in 10 minutes.
+ *
+ * @param {String} loginList
+ */
+function getDomainInfo(loginList) {
+    // If this command fails, we'll retry again in 10 minutes,
+    // arbitrarily chosen giving Bedrock time to resolve the ClearbitCheckPublicEmail job for this email.
+    const RETRY_TIMEOUT = 600000;
+
+    // First we filter out any domains that are in the list of common public domains
+    const emailList = _.filter(loginList, email => (
+        !_.contains(COMMON_PUBLIC_DOMAINS, Str.extractEmailDomain(email))
+    ));
+
+    // If there are no emails left, we have a public domain
+    if (!emailList.length) {
+        Onyx.merge(ONYXKEYS.USER, {isFromPublicDomain: true});
+        return;
+    }
+
+    // Check the API for the remaining uncommon domains
+    API.User_IsFromPublicDomain({emailList: emailList.join(',')})
+        .then((response) => {
+            if (response.jsonCode === 200) {
+                const {isFromPublicDomain} = response;
+                Onyx.merge(ONYXKEYS.USER, {isFromPublicDomain});
+
+                // If the user is not on a public domain we'll want to know whether they are on a domain that has
+                // already provisioned the Expensify card
+                if (isFromPublicDomain) {
+                    return;
+                }
+
+                API.User_IsUsingExpensifyCard()
+                    .then(({isUsingExpensifyCard}) => {
+                        Onyx.merge(ONYXKEYS.USER, {isUsingExpensifyCard});
+                    });
+            } else {
+                // eslint-disable-next-line max-len
+                console.debug(`Command User_IsFromPublicDomain returned error code: ${response.jsonCode}. Most likely, this means that the domain ${Str.extractEmail(sessionEmail)} is not in the bedrock cache. Retrying in ${RETRY_TIMEOUT / 1000 / 60} minutes`);
+                setTimeout(() => getDomainInfo(loginList), RETRY_TIMEOUT);
```

Comment:
It's not great to leave timers running because it can lead to stuff like this function getting called when a user is logged out.

This should help -> https://github.com/Expensify/App/blob/57eed8bd0acdbc41359480fd31311c956d7bab07/src/libs/Timers.js#L11-L27

Reply:
TIL